team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
The `moveGen` stuff is, as mentioned, entirely unnecessary.

`dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0`

is identical to 

`dexEntry.tier.startsWith('Bank') || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0`

because `moveGen` is guaranteed to be `7` when `noPastGenBreeding` is true.